{"absolutePath":"/home/benrai/folder/Audits Code/2025-01-14 silo-v2-cantina-fv_ISSUE/gitmodules/openzeppelin-contracts-5/contracts/utils/Errors.sol","exportedSymbols":{"Errors":[15512]},"id":15513,"license":"MIT","nodeType":"SourceUnit","nodes":[{"id":15497,"literals":["solidity","^","0.8",".20"],"nodeType":"PragmaDirective","src":"33:24:16"},{"abstract":false,"baseContracts":[],"canonicalName":"Errors","contractDependencies":[],"contractKind":"library","documentation":{"id":15498,"nodeType":"StructuredDocumentation","src":"59:254:16","text":" @dev Collection of common custom errors used in multiple contracts\n IMPORTANT: Backwards compatibility is not guaranteed in future versions of the library.\n It is recommended to avoid relying on the error API for critical functionality."},"fullyImplemented":true,"id":15512,"linearizedBaseContracts":[15512],"name":"Errors","nameLocation":"322:6:16","nodeType":"ContractDefinition","nodes":[{"documentation":{"id":15499,"nodeType":"StructuredDocumentation","src":"335:94:16","text":" @dev The ETH balance of the account is not enough to perform the operation."},"errorSelector":"cf479181","id":15505,"name":"InsufficientBalance","nameLocation":"440:19:16","nodeType":"ErrorDefinition","parameters":{"id":15504,"nodeType":"ParameterList","parameters":[{"constant":false,"id":15501,"mutability":"mutable","name":"balance","nameLocation":"468:7:16","nodeType":"VariableDeclaration","scope":15505,"src":"460:15:16","stateVariable":false,"storageLocation":"default","typeDescriptions":{"typeIdentifier":"t_uint256","typeString":"uint256"},"typeName":{"id":15500,"name":"uint256","nodeType":"ElementaryTypeName","src":"460:7:16","typeDescriptions":{"typeIdentifier":"t_uint256","typeString":"uint256"}},"visibility":"internal"},{"constant":false,"id":15503,"mutability":"mutable","name":"needed","nameLocation":"485:6:16","nodeType":"VariableDeclaration","scope":15505,"src":"477:14:16","stateVariable":false,"storageLocation":"default","typeDescriptions":{"typeIdentifier":"t_uint256","typeString":"uint256"},"typeName":{"id":15502,"name":"uint256","nodeType":"ElementaryTypeName","src":"477:7:16","typeDescriptions":{"typeIdentifier":"t_uint256","typeString":"uint256"}},"visibility":"internal"}],"src":"459:33:16"},"src":"434:59:16"},{"documentation":{"id":15506,"nodeType":"StructuredDocumentation","src":"499:89:16","text":" @dev A call to an address target failed. The target may have reverted."},"errorSelector":"d6bda275","id":15508,"name":"FailedCall","nameLocation":"599:10:16","nodeType":"ErrorDefinition","parameters":{"id":15507,"nodeType":"ParameterList","parameters":[],"src":"609:2:16"},"src":"593:19:16"},{"documentation":{"id":15509,"nodeType":"StructuredDocumentation","src":"618:46:16","text":" @dev The deployment failed."},"errorSelector":"b06ebf3d","id":15511,"name":"FailedDeployment","nameLocation":"675:16:16","nodeType":"ErrorDefinition","parameters":{"id":15510,"nodeType":"ParameterList","parameters":[],"src":"691:2:16"},"src":"669:25:16"}],"scope":15513,"src":"314:382:16","usedErrors":[15505,15508,15511],"usedEvents":[]}],"src":"33:664:16"}